### ISSUE TITLE
History "hastag" improvements/issues

### ISSUE THREAD

**Issue by dannon** [galaxyproject/galaxy#3730]
`#hashtags` are a tag that is propagated to derived datasets and displayed alongside history items even when collapsed.

- [x] multiple 'name' tags on a single object
- [x] add tagging support for collections
- [x] tagging needs to redraw parent hda for presentation of nametags
- [x] hashtag for name addition
- [x] tag editor uses a destructive (build-and-replace) render, which breaks dragndrop.
- [x] sort, unique tags display
- [x] when reduce operation is performed (e.g., merging a collection of BAM datasets into a single BAM) tag should propagate from collection to this single resulting dataset
- [ ] enable tagging collection on upload (after upload into collections is done)
- [ ] investigate automatically populating these from dataset name
- [x] investigate use of colors (#7072)
- [x] ability to filter on tags only within tool form (e.g., selecting inputs)
- [ ] in some cases you *don't* want to propagate a tag. The tool XML should allow a setting for disabling propagation

In history collections should display tags just like individual datasets. When collection is expanded:

![coltags](https://cloud.githubusercontent.com/assets/4291636/24165668/61b0e54c-0e47-11e7-8116-492d4d9b4c1b.png)

we need to display tags somewhere around `a list of paired datasets` element.

**Comment by dannon** [comment]
Fixed same-name different-value tag issue preventing tagging in the UI with #3760

**Comment by dannon** [comment]
Tagging for collections in https://github.com/galaxyproject/galaxy/pull/3868

**Comment by jxtx** [comment]
@dannon It looks like tags on collections do not propagate through tool executions

**Comment by dannon** [comment]
Yeah, it looks like where I added the tag addition code, ```handle_output``` only touches the datasets.  I can add it to collections too pretty easily though, it looks like.  Good catch!

**Comment by dannon** [comment]
Redraw also handled in #3868 now.

**Comment by nekrut** [comment]
Testing TAGs on test.galaxyproject.org getting this:
![screen shot 2017-04-15 at 3 32 15 pm](https://cloud.githubusercontent.com/assets/4291636/25066341/223f262c-21f1-11e7-8fe4-1a49371b5c86.png)

Datasets 2 - 5 are labelled with tags. Tags propagate correctly to 6 - 9. In step 10 I merge datasets 6 - 9 into one. Tag A gets duplicated.
This history can be accessed here: https://test.galaxyproject.org/u/anton/h/hiv-r01-example-tagged

Clicking on dataset 10 does not show duplicated tag within tag list:

![screen shot 2017-04-15 at 3 36 53 pm](https://cloud.githubusercontent.com/assets/4291636/25066358/7ca06ef0-21f1-11e7-8fd0-83189a1e05de.png)

**Comment by dannon** [comment]
Hashtags in #3979

**Comment by dannon** [comment]
@nekrut Was the recent addition (sometimes *not* propagating tags) yours?  Can you elaborate?  I'd lean towards doing this in the job execution context (at runtime) rather than doctoring the tool XMLs, if that'd handle your use-case.

**Comment by nekrut** [comment]
@dannon: I can't come up with an example for this

**Comment by dannon** [comment]
Reduction operation tag appending fix should be in #4114

**Comment by dannon** [comment]
Sort/unique done in https://github.com/galaxyproject/galaxy/pull/4139

**Comment by chambm** [comment]
Are workflow-set hashtags supposed to overwrite (reset) the dataset-derived hashtags? If so, is there a way to combine both? I'd like the workflow to propagate which tools and reference genome was used to derived datasets, while the original input dataset hashtags propagate the sample identifier.

**Comment by dannon** [comment]
Resultant tag should be the combination of both, my guess is a bug in the way PJA (workflow set) tags are set is overwriting.   I'll check and get a fix out shortly.

**Comment by yvanlebras** [comment]
Hi everyone, I see another potential issue related to tagging workflow step in "configure output" / "Tags". If at a step X I add a tag #lovegalaxy and that in the step x+1 I add #loveiuc, then executing the wf on a history will result for the x+1 output step on a uniq tag (here #loveiuc). It seems that the previous one is erased...

**Comment by dannon** [comment]
(#4188 fixed @yvanlebras problem)

**Comment by martenson** [comment]
Nothing here seems pressing, removing milestone for now so we do not have to keep pushing it back.

**Comment by dannon** [comment]
Closing — the tags system was completely rewritten with modern Vue components (Tags/, TagsMultiselect/). The old hashtag behavior described here no longer applies.
